### PR TITLE
Aggregates correct zeros

### PR DESCRIPTION
**Pull request by dawidsula26**
Corrected the way zeros are handled in First and Last aggregator.

### CODE REVIEW

File: engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/aggregates.scala [hunk 1]
```diff
@@ -140,24 +140,25 @@ object aggregates {
...
+    override type Aggregate = Option[AnyRef]
 
     override type Element = AnyRef
 
-    override def zero: Aggregate = null
+    override def zero: Aggregate = None
 
-    override def isNeutralForAccumulator(element: LastAggregator.Element, currentAggregate: LastAggregator.Aggregate): Boolean = false
+    override def isNeutralForAccumulator(element: LastAggregator.Element, currentAggregate: LastAggregator.Aggregate): Boolean =
+      currentAggregate.isDefined
```
Comment (mproch) [inline engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/aggregates.scala:151]: Is it? For last it's always false, isn't it?

Comment (dawidsula26) [reply engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/aggregates.scala]: It was wrong. I corrected it and added new tests.

File: engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/aggregates.scala [hunk 2]
```diff
@@ -130,7 +129,7 @@ object aggregates {
 
     override def addElement(el: Element, agg: Aggregate): Aggregate = if (agg.isEmpty) Some(el) else agg
 
-    override def mergeAggregates(agg1: Aggregate, agg2: Aggregate): Aggregate = agg1
+    override def mergeAggregates(agg1: Aggregate, agg2: Aggregate): Aggregate = if (agg1.isEmpty) agg2 else agg1
```
Comment (mproch) [inline engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/aggregates.scala:132]: agg1.getOrElse(agg2)?

File: engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/aggregates.scala [hunk 3]
```diff
@@ -140,24 +139,23 @@ object aggregates {
...
-    override def zero: Aggregate = null
+    override def zero: Aggregate = None
 
     override def isNeutralForAccumulator(element: LastAggregator.Element, currentAggregate: LastAggregator.Aggregate): Boolean = false
 
-    override def addElement(el: Element, agg: Aggregate): Aggregate = el
+    override def addElement(el: Element, agg: Aggregate): Aggregate = Some(el)
 
-    override def mergeAggregates(agg1: Aggregate, agg2: Aggregate): Aggregate = agg2
+    override def mergeAggregates(agg1: Aggregate, agg2: Aggregate): Aggregate = if (agg2.isEmpty) agg1 else agg2
```
Comment (mproch) [inline engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/aggregates.scala:152]: agg2.getOrElse(agg1)?

**Review by mproch — APPROVED** [review]

**Pull request merged**